app/services/spree/afterpay_api/checkout.rb

```diff
@@ -0,0 +1,95 @@
+module Spree::AfterpayApi
+  class Checkout
+    attr_reader :order, :payment, :afterpay_source, :options,
+                :customer, :items, :shipping_address, :billing_address, :discount, :afterpay_order
+
+    def initialize(order, payment, afterpay_source, options = {})
+      @order = order
+      @payment = payment
+      @afterpay_source = afterpay_source
+      @options = options
+      @items = Array.new
+    end
+
+    def perform
+      build_order_params
+      build_afterpay_order
+      send_create_order_request
+    end
+
+    def success?
+      @response && @response.success?
+    end
+
+    private
+
+    def build_order_params
+      build_customer_params
+      build_address_params(:shipping_address)
+      build_address_params(:billing_address)
+      build_line_item_params
+    end
+
+    def build_afterpay_order
+      @afterpay_order = Afterpay::Order.new(
+        total: Money.new(payment.amount * 100, 'AUD'),
+        # total: Money.new(payment.amount * 100, options[:currency]),
+        consumer: customer,
+        items: items,
+        success_url: Spree.railtie_routes_url_helpers.success_afterpay_url(order, host: host_url),
+        cancel_url: Spree.railtie_routes_url_helpers.cancel_afterpay_url(order, host: host_url),
+        reference: order.number,
+        billing_address: billing_address,
+        shipping_address: shipping_address,
+        shipping: Money.new(order.shipment_total * 100, 'AUD')
+        # shipping: Money.new(order.shipment_total * 100, options[:currency])
+      )
+    end
+
+    def send_create_order_request
+      @response = @afterpay_order.create
+      if @response.success?
```

Comment:
what if in case of error

Reply:
No need to perform action as this is handled in success? of same service.